static/app/views/performance/landing/widgets/components/widgetContainer.tsx

```diff
@@ -125,6 +80,9 @@ const _WidgetContainer = (props: Props) => {
       _setChartSetting(index, chartHeight, performanceType, setting);
     }
     setChartSettingState(setting);
+    const newSettings = [...rowChartSettings];
+    newSettings[index] = setting;
+    setRowChartSettings(newSettings);
```

Comment:
Instead of this, can this be lifted to the parent as a helper? Something like

```typescript
function updateRowChartSetting(index, newSetting) {
  // insert the update block you've got here
}
```

and pass that as the prop so that the widget container doesn't need to know about how the row chart settings are stored.